### PR TITLE
[INCL-8] [INCL-9] Adding key methods to controllers

### PR DESCRIPTION
**Pull request by evakule**

### CODE REVIEW

File: src/main/java/com/incl/service/CustomerServiceImpl.java [hunk 1]
```diff
@@ -25,8 +36,34 @@
...
+    return getUsersPhones(usersList, interestAreaEntity);
+  }
+  
+  private List<String> getUsersPhones(
+      List<UserEntity> usersList,
+      InterestAreaEntity interestAreaEntity
+  ) {
+    return usersList.stream()
+        .filter(userEntity -> userEntity
+            .getInterests()
```
Comment (ssuchkov) [inline src/main/java/com/incl/service/CustomerServiceImpl.java]: can user have no interests?

File: src/main/java/com/incl/service/CustomerServiceImpl.java [hunk 2]
```diff
@@ -25,8 +36,34 @@
...
+            .contains(interestAreaEntity))
+        .map(UserEntity::getPhone)
```
Comment (ssuchkov) [inline src/main/java/com/incl/service/CustomerServiceImpl.java]: can user have no phone?

File: src/main/java/com/incl/service/CustomerServiceImpl.java [hunk 3]
```diff
@@ -25,8 +36,34 @@
...
+    CountryEntity countryEntity =
+        countryRepository.getOne(countryId);
+    InterestAreaEntity interestAreaEntity =
+        interestAreaRepository.getOne(interestId);
+    List<UserEntity> usersList = userRepository
+        .getUsersByCountry(countryEntity);
+    return getUsersPhones(usersList, interestAreaEntity);
+  }
+  
+  private List<String> getUsersPhones(
```
Comment (ssuchkov) [inline src/main/java/com/incl/service/CustomerServiceImpl.java]: why don't you make everything in single query?

File: src/main/java/com/incl/service/InterestAreaServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  
+  @Override
+  public UserInterestAreaDTO getByMinPosts() {
+    List<InterestAreaEntity> areaEntityList =
+        interestAreaRepository.findAll();
+    InterestAreaEntity interestAreaEntity = getByMinimumPosts(areaEntityList);
+    return userInterestAreaMapper.toDto(interestAreaEntity);
+  }
+  
+  private InterestAreaEntity getByMaxUsers(final List<InterestAreaEntity> list) {
```
Comment (ssuchkov) [inline src/main/java/com/incl/service/InterestAreaServiceImpl.java]: not sure if the best solution fetch everything and traverse then

Comment (evakule) [reply src/main/java/com/incl/service/InterestAreaServiceImpl.java]: It`s ok

File: src/main/java/com/incl/service/CustomerServiceImpl.java (hunk 3 above)
Comment (evakule) [reply src/main/java/com/incl/service/CustomerServiceImpl.java]: It`s ok

File: src/main/java/com/incl/service/CustomerServiceImpl.java (hunk 1 above)
Comment (evakule) [reply src/main/java/com/incl/service/CustomerServiceImpl.java]: Yes, he can

File: src/main/java/com/incl/service/CustomerServiceImpl.java (hunk 2 above)
Comment (evakule) [reply src/main/java/com/incl/service/CustomerServiceImpl.java]: Ok

**Pull request merged**